tools/GdkTestRunner/Modules/ModuleLibrary.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json.Linq;
+using NLog;
+
+namespace GdkTestRunner.Modules
+{
+    public static class ModuleLibrary
+    {
+        public static List<string> LoadedModuleIdentifiers => moduleTypeLibrary.Keys.ToList();
+
+        private static readonly Dictionary<string, Type> moduleTypeLibrary = new Dictionary<string, Type>();
+        private static readonly Dictionary<string, string> moduleDocumentation = new Dictionary<string, string>();
+
+        static ModuleLibrary()
+        {
+            var types = AppDomain.CurrentDomain.GetAssemblies().SelectMany(assembly => assembly.GetTypes());
+
+            var moduleTypes = types.Where(type => typeof(BaseModule).IsAssignableFrom(type) && !type.IsAbstract);
+
+            foreach (var moduleType in moduleTypes)
+            {
+                var identifierAttribute = Attribute.GetCustomAttribute(moduleType,
+                    typeof(TestModuleIdentifierAttribute)) as TestModuleIdentifierAttribute;
+
+                if (identifierAttribute == null)
+                {
+                    LogManager.GetCurrentClassLogger().Error($"Encountered subclass of {typeof(BaseModule).FullName} " +
+                        $"without {typeof(TestModuleIdentifierAttribute).FullName}");
```

Comment:
what's the difference between `typeof(TestModuleIdentifierAttribute).FullName` and `nameof(TestModuleIdentifierAttribute)`?

Reply:
Not sure - I think `nameof` might be compile-time